ddtrace/appsec/processor.py

```diff
@@ -93,7 +93,8 @@ def _set_headers(span, headers):
     # type: (Span, Dict) -> None
     for k in headers:
         if k.lower() in _COLLECTED_REQUEST_HEADERS:
-            span._set_str_tag(_normalize_tag_name("request", k), headers[k])
+            # since the header value can now be a list, we can't write it as a str tag directly
```

Comment:
```suggestion
            # since the header value can be a list, use `set_tag()` to ensure it is converted to a string
```

we should at least do:

```python
def _set_headers(span, headers):
    # type: (Span, Dict[str, Union[str, List[str]) -> None
```

then it should have gotten picked up by type hinting?

Reply:
yes indeed!